src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -40,25 +41,120 @@ class ThetaStarPathPlanner : public PathPlanner
                                  const std::vector<ObstaclePtr> &obstacles) override;
 
    private:
-    class Coordinate : public std::pair<unsigned int, unsigned int>
+    class Coordinate
     {
        public:
         Coordinate(unsigned int row, unsigned int col)
-            : std::pair<unsigned int, unsigned int>(row, col)
+            : row_(row), col_(col), internal_comparison_key_(computeComparisonKey(*this))
         {
+            assert(row < (1 << 16));
+            assert(col < (1 << 16));
         }
 
-        Coordinate() : std::pair<unsigned int, unsigned int>() {}
+        Coordinate() : row_(0), col_(0) {}
 
         unsigned int row(void) const
         {
-            return this->first;
+            return row_;
         }
 
         unsigned int col(void) const
         {
-            return this->second;
+            return col_;
         }
+
+        unsigned int internalComparisonKey(void) const
+        {
+            return internal_comparison_key_;
+        }
+        bool operator<(const Coordinate &other) const
+        {
+            return internal_comparison_key_ < other.internal_comparison_key_;
+        }
+
+        bool operator==(const Coordinate &other) const
+        {
+            return internal_comparison_key_ == other.internal_comparison_key_;
+        }
+
+       private:
+        /**
+         * Calculate the key of this coordinate based on row and col for comparison. row
+         * and col are smaller than 1<<16 (65536) row value will occupy low 16 bits and
+         * col values will occupy high 16 bits in key
+         *
+         * @param coord the reference to this coordinate itself
+         * @return the key value in unsigned int
+         */
+
+        static unsigned int computeComparisonKey(const Coordinate &coord)
+        {
+            return coord.row() + coord.col() * (1 << 16);
+        }
+
+        unsigned int row_;
+        unsigned int col_;
+        unsigned int internal_comparison_key_;
```

Comment:
A quick comment on what this member is would be helpful.

Reply:
that makes sense

Reply:
bumping this

Reply:
comment added